pkgs/build-support/go/module.nix

```diff
@@ -85,6 +87,9 @@ let
       runHook preConfigure
       export GOCACHE=$TMPDIR/go-cache
       export GOPATH="$TMPDIR/go"
+      # fixes 'GOPROXY list is not the empty string, but contains no entries'
+      # "https://proxy.golang.org,direct" is the go default
+      export GOPROXY="''${GOPROXY:-"https://proxy.golang.org,direct"}" # respect impureEnvVars
```

Comment:
I am also receiving ``github.com/stretchr/testify/assert: github.com/stretchr/testify@v1.8.4: verifying go.mod: missing GOSUMDB`` unless I have ``GOSUMDB=off`` env set. Not sure if this is related but it feels odd.

Reply:
> If I don't set this variable I am always receiving the following error when vendoring something with go: [...] `GOPROXY list is not the empty string, but contains no entries`

Just in case that's in response to me, and/or just to provide another data point... This is the main one I had to set to get my "normal" go packages building again.

Reply:
probably better setting GOSUMDB to the default? (sum.golang.org)

Reply:
we just need to install the `go.env` into GOROOT.
https://github.com/golang/go/blob/master/go.env
https://github.com/golang/go/issues/57179

Reply:
> we just need to install the `go.env` into GOROOT. https://github.com/golang/go/blob/master/go.env [golang/go#57179](https://github.com/golang/go/issues/57179)

fixed: https://github.com/NixOS/nixpkgs/pull/250472